tests/common-unit/hierarchical-menu/optimistic-ui.ts

```diff
@@ -0,0 +1,118 @@
+import { wait } from '@instantsearch/testutils';
+import {
+  createSearchClient,
+  createMultiSearchResponse,
+  createSingleSearchResponse,
+} from '@instantsearch/mocks';
+import type { HierarchicalMenuSetup } from '.';
+import { fakeAct } from '../common';
+import userEvent from '@testing-library/user-event';
+
+export function createOptimisticUiTests(setup: HierarchicalMenuSetup) {
+  // @TODO: after helper is updated with https://github.com/algolia/algoliasearch-helper-js/pull/925, enable this test
+  describe.skip('optimistic UI', () => {
+    test('is checked immediately with a slow network', async () => {
+      const delay = 100;
+      const margin = 10;
+      const attributes = ['brand'];
+      const options = {
+        instantSearchOptions: {
+          indexName: 'indexName',
+          searchClient: createSearchClient({
+            search: jest.fn(async (requests) => {
+              await wait(delay);
+              return createMultiSearchResponse(
+                ...requests.map(() =>
+                  createSingleSearchResponse({
+                    facets: {
+                      [attributes[0]]: {
+                        Samsung: 100,
+                        Apple: 1000,
+                      },
+                    },
+                  })
+                )
+              );
+            }),
+          }),
+        },
+        attributes,
+      };
+      const env = await setup(options);
+      const { act = fakeAct } = env;
+
+      // wait for initial results
+      await act(async () => {
+        await wait(margin + delay);
+        await wait(0);
```

Comment:
Could you explain or comment why we wait a second time after most of the interactions?

Reply:
I don't fully understand, but I believe it's related to jest timers being quite strict in ordering. If there's no two timers waited for, a timer added _during_ the initial wait will still be triggered after.